File: largest-number-at-least-twice-of-others/largest-number-at-least-twice-of-others.py
class Solution(object):
    def dominantIndex(self, nums):
        maxnum = max(nums)
        maxindex = nums.index(max(nums))
        nums.pop(maxindex)
        counter = 0
        for i in nums:
            if(i*2<=maxnum):
                counter+=1
            else:
                counter = 0
                break
                
        if counter>0:
            return maxindex
        else:
            return -1;
    
        
        

File: largest-number-at-least-twice-of-others/largest-number-at-least-twice-of-others.cpp
class Solution {
public:
    int dominantIndex(vector<int>& nums) {
        int maxindex = max_element(nums.begin(), nums.end()) - nums.begin();
        int maxnum = *max_element(nums.begin(), nums.end());
        sort(nums.begin(),nums.end());
        nums.pop_back();
        for(int i =0; i<nums.size();i++){
            if(maxnum<nums[i]*2)
                return -1;
        }
        return maxindex;
    }
};